server/eclipse-project/src/main/java/us/freeandfair/corla/query/ContestResultQueries.java

```diff
@@ -0,0 +1,40 @@
+/** Copyright (C) 2018 the Colorado Department of State  **/
+
+package us.freeandfair.corla.query;
+
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.hibernate.Session;
+
+import us.freeandfair.corla.model.ContestResult;
+import us.freeandfair.corla.persistence.Persistence;
+
+
+public final class ContestResultQueries {
+  /**
+   * prevent construction
+   */
+  private ContestResultQueries() {
+  }
+
+  /**
+   * Return the ContestResult with the contestName given or create a new
+   * ContestResult with the contestName.
+   **/
+  public static ContestResult findOrCreate(final String contestName) {
+    final Session s = Persistence.currentSession();
+    final Query q = s.createQuery("select cr from ContestResult cr " +
+                                  "where cr.contestName = :contestName");
+    q.setParameter("contestName", contestName);
+    final List<ContestResult> results = q.getResultList();
```

Comment:
Do you think [uniqueResultOptional](http://docs.jboss.org/hibernate/orm/5.2/javadocs/org/hibernate/query/Query.html#uniqueResultOptional--) would make for less code here? It raises an exception if there is more than one result, but since `contestName` is unique among `ContestResult`s, that seems like it shouldn't happen unless we screwed something up.